androidApp/src/main/java/com/example/tasky/android/agenda/di/AgendaModule.kt

```diff
@@ -3,13 +3,14 @@ package com.example.tasky.android.agenda.di
 import com.example.tasky.agenda.data.AgendaRepository
 import com.example.tasky.agenda.domain.IAgendaRepository
 import com.example.tasky.android.agenda.viewmodel.AgendaViewModel
-import org.koin.core.module.dsl.singleOf
 import org.koin.core.module.dsl.viewModelOf
 import org.koin.dsl.bind
 import org.koin.dsl.module
 
 val agendaModule =
     module {
-        singleOf<AgendaRepository>(::AgendaRepository).bind<IAgendaRepository>()
+        single<AgendaRepository> {
+            AgendaRepository(get(), get())
+        }.bind<IAgendaRepository>()
```

Comment:
`None of the following functions can be called with the arguments supplied.`

![image](https://github.com/user-attachments/assets/66bafad2-8ee5-413d-94f7-4614834096d0)

Reply:
The error occurs since I changed the constructor of AgendaRepository from

```
class AgendaRepository(
    private val agendaDataSource: AgendaDataSource = AgendaDataSource(),
    private val agendaLocalDataSource: AgendaLocalDataSource = AgendaLocalDataSource(),
    private val konnection: Konnection = Konnection.instance,
)
```

to

```
class AgendaRepository(
    private val alarmScheduler: IAlarmScheduler,
    private val alarmRepository: IAlarmRepository,
    private val agendaDataSource: AgendaDataSource = AgendaDataSource(),
    private val agendaLocalDataSource: AgendaLocalDataSource = AgendaLocalDataSource(),
    private val konnection: Konnection = Konnection.instance,
)
```

Reply:
No, there is only 1 `AgendaRepository` class definition.

Same error occurs if I change the type to `IAgendaRepository`.

If I omit the type, the error would be `Overload resolution ambiguity`
![image](https://github.com/user-attachments/assets/1e47a981-e4de-4495-a63b-f80d8aaeb520)

Reply:
That error is thrown because of the default parameters you have inside `AgendaRepository`. Those with default parameters are all concrete implementations. In this situation, I would suggest creating interfaces in your domain layers for those `DataSource` classes and inject them via Koin. I would reference the singleton, `Konnection`, in a local variable.